### PR TITLE
Add show/hide tests

### PR DESCRIPTION
**Pull request by eliagrady**

### CODE REVIEW

File: src/lib/show-hide.spec.js [hunk 1]
```diff
@@ -1,30 +1,227 @@
+/* global sinon, expect, describe, it, beforeEach, afterEach */
+/* eslint-disable noinspection, prefer-arrow-callback, func-names, no-underscore-dangle */
 import { show, hide, __RewireAPI__ } from './show-hide';
 
+const jsdom = require('jsdom-global');
```
Comment (nadavsinai) [inline src/lib/show-hide.spec.js:5]: this should not be needed anymore, jsDom is not global, please remove this and the beforeEach/afterEach blocks below

File: src/lib/show-hide.spec.js [hunk 2]
```diff
@@ -1,30 +1,227 @@
...
+  it('calls dispatchEvent on the wrapper with type `dialog:show-before` and dialog wrapper as data',
+    function () {
+      setup({ dispatchRetValue: false });
+      show.call(dialogDummy);
+      expect(myDispatch).to.have.been.calledOnce();
+      expect(myDispatch).to.have.been.calledWith(
+        dialogDummy.wrapper, 'dialog:show-before', { dialog: dialogDummy.wrapper });
+    });
+  it('cancels show (does not call show-after or goToDialog ) if event was prevented', function () {
+    setup({ dispatchRetValue: false });
```
Comment (nadavsinai) [inline src/lib/show-hide.spec.js:34]: oups, you're not calling show at all here, call it and assert that show-after is not called (not show-before)

File: src/lib/show-hide.spec.js [hunk 3]
```diff
@@ -1,30 +1,227 @@
...
+    setup({ dispatchRetValue: true });
+    show.call(dialogDummy);
+    expect(dialogDummy.isVisible).to.be.true();
+  });
+  it('removes the aria-hidden attribute on the wrapper', function () {
+    setup({ dispatchRetValue: true });
+    show.call(dialogDummy);
+    expect(dialogDummy.wrapper.getAttribute('aria-hidden')).not.to.exist();
+  });
+  it('keeps track of focuses element on focusOnClose property on the dialogState ', function () {
```
Comment (nadavsinai) [inline src/lib/show-hide.spec.js:49]: sorry my mistake - the test could be made less brittle by checking the complete behavior

keep this test -- but also add one like this :

focus an element (as you did)
call show (assert element is not focused)
call hide 
assert element is back in focus

this way we don't care for internal implementation (which may change) and check the actual behavior int he spec.
this is a little more than unit test (as it calls two functions) and I would move it from the show/hide spec into a separate "intergation specs" file

File: src/lib/show-hide.spec.js [hunk 4]
```diff
@@ -1,30 +1,227 @@
...
+  });
+  it('sets aria-hidden on elemToConceal', function () {
+    setup({ dispatchRetValue: true });
+    show.call(dialogDummy);
+    expect(dialogDummy.elemToConceal.getAttribute('aria-hidden')).to.equal('true');
+  });
+  it('adds focus eventListener on the document.body calling hideWhenFocusLost', function () {
+    setup({ dispatchRetValue: true });
+    show.call(dialogDummy);
+    expect(addEventListenerStub).to.have.been.calledWith(
```
Comment (nadavsinai) [inline src/lib/show-hide.spec.js:73]: this addEventStub spy  is not created  here - it should not rely on other tests

File: src/lib/show-hide.spec.js [hunk 5]
```diff
@@ -1,30 +1,227 @@
...
+    function () {
+      setup({ dispatchRetValue: false });
+      hide.call(dialogDummy);
+      expect(myDispatch).to.have.been.calledOnce();
+      expect(myDispatch).to.have.been.calledWith(
+        dialogDummy.wrapper, 'dialog:hide-before', { dialog: dialogDummy.wrapper });
+    });
+  it('cancels show (does not call show-after or goToDialog ) if event was prevented', function () {
+    setup({ dispatchRetValue: false });
+    expect(dialogDummy.goToDialog).to.have.been.not.called();
```
Comment (nadavsinai) [inline src/lib/show-hide.spec.js:146]: same as before - you need to call hide here

File: .gitignore [hunk 6]
```diff
@@ -15,3 +15,6 @@ jspm_packages
 /coverage
 /.tmp
 .nyc_output
+
+# Ignore IDE files
+/.idea
```
Comment (nadavsinai) [inline .gitignore:20]: good stuff - I just add this in my computer global gitignore - i suggest you do the same

File: src/lib/show-hide.spec.js (hunk 3 above)
Comment (eliagrady) [reply src/lib/show-hide.spec.js:49]: this test uses both show and hide, so it's place should be on neither describe blocks - it should be a third describe block for full flow tests.

Comment (nadavsinai) [reply src/lib/show-hide.spec.js:49]: ok

**Comment by nadavsinai** [comment]
Hi, can you please re-open this against master ?

**Pull request closed without merging**